Observer/BeforeShipmentObserver.php

```diff
@@ -80,13 +80,13 @@ public function execute(Observer $observer)
             return;
         }
 
-        $captureOnShipment = $this->configHelper->getConfigData(
-            'capture_on_shipment',
+        $openInvoiceCapture = $this->configHelper->getConfigData(
+            'capture_for_openinvoice',
             self::XML_ADYEN_ABSTRACT_PREFIX,
             $order->getStoreId()
         );
 
-        if (!$captureOnShipment) {
+        if ($openInvoiceCapture != 'onshipment') {
```

Comment:
Can you please use `strcmp` and move the string to constants preferably the same place with the first comment.